modules/animation/src/datastructures/animation.cpp

```diff
@@ -266,6 +266,40 @@ Seconds Animation::getLastTime() const {
     return time;
 }
 
+Seconds Animation::getPrevTime(Seconds at) const {
+    auto prevKeyframeTime = at;
```

Comment:
better to use a optional here to track the "unset" state then checking if it equal to at... that also propagates better upwards....